monitor/validators.js

```diff
@@ -89,19 +105,40 @@ async function main(bridgeMode) {
   })
   const foreignVBalances = {}
   const homeVBalances = {}
-  logger.debug('calling getGasPrices')
-  const gasPriceInGwei = await getGasPrices(GAS_PRICE_SPEED_TYPE)
-  const gasPrice = new Web3Utils.BN(Web3Utils.toWei(gasPriceInGwei.toString(10), 'gwei'))
-  const txCost = gasPrice.mul(new Web3Utils.BN(GAS_LIMIT))
+
+  logger.debug('calling home getGasPrices')
+  const homeGasPrice = await getGasPrices(
+    HOME_GAS_PRICE_ORACLE_URL,
+    HOME_GAS_PRICE_SPEED_TYPE,
+    HOME_GAS_PRICE_FACTOR,
+    HOME_GAS_PRICE_FALLBACK
+  )
+  const homeGasPriceGwei = Web3Utils.fromWei(homeGasPrice.toString(), 'gwei')
+  const homeTxCost = homeGasPrice.mul(Web3Utils.toBN(HOME_GAS_LIMIT))
+
+  logger.debug('calling foreign getGasPrices')
+  const foreignGasPrice = await getGasPrices(
+    FOREIGN_GAS_PRICE_ORACLE_URL,
+    FOREIGN_GAS_PRICE_SPEED_TYPE,
+    FOREIGN_GAS_PRICE_FACTOR,
+    FOREIGN_GAS_PRICE_FALLBACK
+  )
+  const foreignGasPriceGwei = Web3Utils.fromWei(foreignGasPrice.toString(), 'gwei')
+  const foreignTxCost = foreignGasPrice.mul(Web3Utils.toBN(FOREIGN_GAS_LIMIT))
+
   let validatorsMatch = true
   logger.debug('calling asyncForEach foreignValidators foreignVBalances')
   await asyncForEach(foreignValidators, async v => {
     const balance = await web3Foreign.eth.getBalance(v)
-    const leftTx = new Web3Utils.BN(balance).div(txCost).toString(10)
+    const leftTx = foreignTxCost.isZero()
```

Comment:
Is `foreignTxCost.isZero()` possible? Even if so should `leftTx` be zero in this case? Does it make sense to report a very big number or something like `infinite`?

Reply:
You are right, I don't think zero is possible on foreign. Removed that check for foreign. On home side, if `homeTxCost` is `zero`, `leftTx` will be `999999`, I tried to use `Infinity` number, but it does not play well with `JSON.stringify` to write the results. https://github.com/poanetwork/tokenbridge/pull/142/commits/b22a0a2a22de0524a47e5cad4c7aa134d021891c